File: src/dataset/go/food_chain_Exercism_go_drapala/gemini/GeminiFlash_food_chain_v3.go
package foodchain

import (
	"fmt"
	"strings"
)

type verse struct {
	animal  string
	comment string
}

var verses = []verse{
	{"fly", ""},
	{"spider", "It wriggled and jiggled and tickled inside her."},
	{"bird", "How absurd to swallow a bird!"},
	{"cat", "Imagine that, to swallow a cat!"},
	{"dog", "What a hog, to swallow a dog!"},
	{"goat", "Just opened her throat and swallowed a goat!"},
	{"cow", "I don't know how she swallowed a cow!"},
	{"horse", "She's dead, of course!"},
}

var catchPhrases = []string{
	"",
	"She swallowed the spider to catch the fly that wriggled and jiggled and tickled inside her.",
	"She swallowed the bird to catch the spider.",
	"She swallowed the cat to catch the bird.",
	"She swallowed the dog to catch the cat.",
	"She swallowed the goat to catch the dog.",
	"She swallowed the cow to catch the goat.",
}

func Verse(v int) string {
	v--
	if v < 0 || v >= len(verses) {
		return ""
	}

	var sb strings.Builder
	sb.WriteString(fmt.Sprintf("I know an old lady who swallowed a %s.\n", verses[v].animal))
	sb.WriteString(verses[v].comment)

	if v < len(catchPhrases) && v > 0 {
		for i := v; i > 0; i-- {
			sb.WriteString(fmt.Sprintf("\n%s", catchPhrases[i]))
		}
		sb.WriteString("\n")
	}

	if v != 7 {
		sb.WriteString("I don't know why she swallowed the fly. Perhaps she'll die.")
	}

	return sb.String()
}

func Verses(start, end int) string {
	var sb strings.Builder
	for i := start; i <= end; i++ {
		sb.WriteString(Verse(i))
		if i < end {
			sb.WriteString("\n\n")
		}
	}
	return sb.String()
}

func Song() string {
	return Verses(1, 8)
}

File: src/dataset/cpp/food_chain_Exercism_thefullarcticfox/gemini/GeminiFlash_food_chain_v3.cpp
#include "food_chain.h"
#include <array>
#include <string>

namespace food_chain {

const std::array<std::pair<std::string_view, std::string_view>, 8> versemap = {
    std::make_pair("fly", "I don't know why she swallowed the fly. Perhaps she'll die."),
    std::make_pair("spider", "It wriggled and jiggled and tickled inside her."),
    std::make_pair("bird", "How absurd to swallow a bird!"),
    std::make_pair("cat", "Imagine that, to swallow a cat!"),
    std::make_pair("dog", "What a hog, to swallow a dog!"),
    std::make_pair("goat", "Just opened her throat and swallowed a goat!"),
    std::make_pair("cow", "I don't know how she swallowed a cow!"),
    std::make_pair("horse", "She's dead, of course!")
};

std::string verse(int verse_num) {
    std::string res;
    res.reserve(256); 

    const auto& [animal, phrase] = versemap[verse_num - 1];
    res += "I know an old lady who swallowed a ";
    res += animal;
    res += ".\n";

    if (verse_num > 1) {
        res += phrase;
        res += "\n";
    }

    if (verse_num == 8) {
        return res;
    }

    int current_verse = verse_num;
    while (current_verse > 1) {
        res += "She swallowed the ";
        res += versemap[current_verse - 1].first;
        --current_verse;
        res += " to catch the ";
        res += versemap[current_verse - 1].first;

        if (current_verse == 2) {
            res += " that wriggled and jiggled and tickled inside her";
        }
        res += ".\n";
    }

    res += versemap[0].second;
    res += "\n";
    return res;
}

std::string verses(int first, int last) {
    std::string res;
    res.reserve((last - first + 1) * 256);

    for (int i = first; i <= last; ++i) {
        res += verse(i);
        if (i < last) {
            res += "\n";
        }
    }
    return res;
}

std::string sing() {
    return verses(1, 8);
}

}  // namespace food_chain